cli/azd/pkg/project/service_target_containerapp.go

```diff
@@ -85,19 +88,29 @@ func (at *containerAppTarget) Deploy(
 				return
 			}
 
-			// Login, tag & push container image to ACR
-			containerDeployTask := at.containerHelper.Deploy(ctx, serviceConfig, packageOutput, targetResource, true)
-			syncProgress(task, containerDeployTask.Progress())
-
-			_, err := containerDeployTask.Await()
-			if err != nil {
-				task.SetError(err)
-				return
+			if serviceConfig.Docker.RemoteBuild {
+				err := at.remoteBuildHelper.RunRemoteBuild(ctx, serviceConfig, targetResource, func(s string) {
+					task.SetProgress(NewServiceProgress(s))
+				})
+				if err != nil {
+					task.SetError(fmt.Errorf("running remote build: %w", err))
+					return
+				}
+			} else {
+				// Login, tag & push container image to ACR
+				containerDeployTask := at.containerHelper.Deploy(ctx, serviceConfig, packageOutput, targetResource, true)
+				syncProgress(task, containerDeployTask.Progress())
+
+				_, err := containerDeployTask.Await()
+				if err != nil {
+					task.SetError(err)
+					return
+				}
 			}
```

Comment:
What if we moved the concern of checking whether remote build is enabled to inside of `containerHerlper.Deploy(...)`. Then both ACA & AKS will inherently be able to take advantage of it instead of checking directly within each service target?